### PR TITLE
Adds docs and Fixes Imports for ops and io

### LINKED ISSUE #2649: Docs of some functions written are missing
## 📚 Documentation

A simple issue, Docs are missing on the torchvision website for following functions written in torchvision.

I guess we should add these docs on the webpage, as end-users will benefit from using these functions. 

Most people will not look at source code to find these functions but refer to docs.

Missing docs that I found

- [ ] Image reading functions [here](https://github.com/pytorch/vision/blob/master/torchvision/io/image.py)
We have docs for video io functions, so maybe image should too be there.

- [ ] Torchvision ops from [boxes.py](https://github.com/pytorch/vision/blob/master/torchvision/ops/boxes.py). Docs are added for NMS. but we are missing IoU, Box area and some classes. Partly fixed in #2642 

Please do let me know if some other docs or missing as well.

Also, I can raise a PR to fix these, please do let me know if it is needed!

### PR DESCRIPTION
**Pull request by oke-aditya**
Closes #2649 .

Adds docs for 

- [x] Image reading functions [here](https://github.com/pytorch/vision/blob/master/torchvision/io/image.py)

- [x] Torchvision ops from [boxes.py](https://github.com/pytorch/vision/blob/master/torchvision/ops/boxes.py).

Some functions were missing in the `__init__.py` files, I have added those imports too. 
Please review and do let me know if changes are needed.

cc @vfdev-5 @fmassa @pmeier

### CODE REVIEW

**Review by pmeier — APPROVED** [review]

**Comment by vfdev-5** [comment]
Hi @oke-aditya , sorry for delay, we discussed with @fmassa about this PR and we think that exposing `decode_*/read_*` from io module is not suitable as in future the idea is to provide a bit more generic functions like `read` and `decode` which could dispatch to those `decode_*/read_*` functions. Thus, user does not need to check the format and dispatch in his/her code.

For torchvision ops, they were not included in 0.3.0 release, but now we may consider them as more mature and make them public. Let me comment what we would to remove and we can merge this PR.

File: docs/source/io.rst [hunk 1]
```diff
@@ -6,6 +6,17 @@ torchvision.io
 The :mod:`torchvision.io` package provides functions for performing IO
 operations. They are currently specific to reading and writing video.
 
+Image
```
Comment (vfdev-5) [inline docs/source/io.rst:9]: Let's revert this modification.

File: torchvision/io/__init__.py [hunk 2]
```diff
@@ -15,6 +15,12 @@
     write_video,
 )
 
+from .image import (
```
Comment (vfdev-5) [inline torchvision/io/__init__.py:18]: Let's revert this import and `__all__` modification

**Review by vfdev-5 — COMMENTED** [review]
@oke-aditya I left few comment. If you could handle that it would be perfect and we can merge this PR after that. Thanks !

**Comment by oke-aditya** [comment]
Ok @vfdev-5 sure, I will undo the changes from `io` module. I saw other PR being opened to revamp torchvision video reader and other APIs #2660 . 

I will do changes as suggested 👍 and ask for a review once again

File: docs/source/io.rst (hunk 1 above)
Comment (oke-aditya) [reply docs/source/io.rst:9]: reverted

File: torchvision/io/__init__.py (hunk 2 above)
Comment (oke-aditya) [reply torchvision/io/__init__.py:18]: reverted

**Review by vfdev-5 — APPROVED** [review]
LGTM ! Thanks @oke-aditya !

**Pull request merged**